### DISCUSSION TITLE
non-standard properties on Function prototype

### DISCUSSION THREAD

**Discussion by p-bakker** [quickjs-ng/quickjs#713] [Q&A]
Looks like there are some non-standard properties on the function prototype, like columnNumber, lineNumber and fileName

Do they serve a purpose?

**Comment by p-bakker** [comment]
And Set seems to have a non-standard .groupBy method

**Reply by saghul** [reply]
This is a bug because Set and Map reuse the class functions, we can fix that, thanks for the report!

**Comment by saghul** [comment]
> Looks like there are some non-standard properties on the function prototype, like columnNumber, lineNumber and fileName
> 
> Do they serve a purpose?

Lol I had missed QuickJS has those. I see they code is very intentional, but there doesn't seem to be a good reason for them to be exposed to JS I guess? @bnoordhuis thoughts?

**Reply by p-bakker** [reply]
Discovered those through SES lockdown, as it logs and removes anything that is non-standard (and the SES code already had quite a few 'exceptions' paths in their codebase to remove stuff the initial quickjs implementation has.

After commenting out the quickjs-related exceptions in the SEs codebase, I found the non-standard properties remaining in quickJS-NG were the ones mentioned above

**Reply by saghul** [reply]
👍

**Reply by saghul** [reply]
https://github.com/quickjs-ng/quickjs/pull/716
